ReportingServicesTools/Functions/CatalogItems/Rest/Write-RsRestCatalogItem.ps1

```diff
@@ -224,6 +228,74 @@ function Write-RsRestCatalogItem
                 $payload = ConvertFrom-Json $content
                 $payload.Path = $itemPath
             }
+            elseif ($itemType -eq "PowerBIReport")
+            {
+                $fileBytes = [System.IO.File]::ReadAllBytes($EntirePath)
+                $fileSizeInMb = (Get-Item $EntirePath).length/1MB
+
+                if ($fileSizeInMb -ge $minPortalLargeFileSizeInMb)
+                {
+                    $maxServerFileSizeInMb = Get-RsRestPublicServerSetting -Property "MaxFileSizeMb" -ReportPortalUri $ReportPortalUri -WebSession $WebSession
+                    if ($fileSizeInMb -gt $maxFileSizeInMb)
+                    {
+                        throw "This file is too large to be uploaded. Files larger than $maxFileSizeInMb MB are not currently supported: $item!"
+                    }
+                    elseif ($maxServerFileSizeInMb -gt 0 -and $fileSizeInMb -gt $maxServerFileSizeInMb) {
+                        throw "This file is too large to be uploaded. Files larger than $maxServerFileSizeInMb MB are not currently supported: $item!"
+                    }
+
+                    Write-Verbose "PowerBIReport $item is a large"
+
+                    $isLargePowerBIReport = $true
+                    $pbixPayload = [System.Text.Encoding]::GetEncoding('ISO-8859-1').GetString($fileBytes)
+                    $boundary = [System.Guid]::NewGuid().ToString()
+                    $LF = "`r`n"
+
+                    $bodyLines = (
+                        # Name
+                        "--$boundary",
+                        "Content-Disposition: form-data; name=`"Name`"$LF",
+                        $itemName,
+                        # ContentType
+                        "--$boundary",
+                        "Content-Disposition: form-data; name=`"ContentType`"$LF",
+                        "",
+                        # Content
+                        "--$boundary",
+                        "Content-Disposition: form-data; name=`"Content`"$LF",
+                        "undefined",
+                        # Path
+                        "--$boundary",
+                        "Content-Disposition: form-data; name=`"Path`"$LF",
+                        $itemPath,
+                        # @odata.type
+                        "--$boundary",
+                        "Content-Disposition: form-data; name=`"@odata.type`"$LF",
+                        "#Model.PowerBIReport",
+                        # File
+                        "--$boundary",
+                        "Content-Disposition: form-data; name=`"File`"; filename=`"$itemName`"",
+                        "Content-Type: application/octet-stream$LF",
+                        $pbixPayload,
+                        "--$boundary--"
+                    ) -join $LF
+                }
+                else
+                {
+                    Write-Verbose "PowerBIReport $item is a small"
```

Comment:
this code is duplicated of the next else.

Please collapse the condition from line 236 into line 231 so you don't need nested ifs